### ISSUE TITLE
Reindexing blockchain fails without notice

### ISSUE THREAD

**Issue by osnwt** [graphprotocol/graph-node#1568]
I have issues rescanning blockchain with new database for successfully deployed contract. Steps are:

- Created simple ERC20 token in Rinkeby network. Successfully deployed it. Graph scanned the network using infura.io provider, all works as expected. All handler functions work.
- Simulated new Graph instance creation: stopped Graph, IPFS and Postgres. Removed data files. Restarted everything.
- Created the same subgraph. Deployed the same subgraph. No errors reported.
- Graph starts scanning the chain, takes one event and does nothing anymore. The event handler function called but never exited according to logs. Exactly the same code worked initially.

Relevant code fragments (minimal):

```
export function handleMint(event: Mint): void {
  log.warning('handleMint: step 1', [])           // <- logged
  let contract = Contract.bind(event.address)
  log.warning('handleMint: step 2', [])           // <- logged
  let owner = contract.owner()
  log.warning('handleMint: step 3', [])           // <- NEVER LOGGED
  return
}
```
```
contract Ownable {
    address public owner;    
    constructor() public {
        owner = msg.sender;
    }
    ...
}

contract SimpleTokenCoin is Ownable {
    ...
}
```
```
graph-node    | Mar 30 17:26:51.898 INFO Received subgraph_deploy request, params: SubgraphDeployParams { name: SubgraphName("os/stc"), ipfs_hash: SubgraphDeploymentId("QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA"), node_id: None }, component: JsonRpcServer
graph-node    | Mar 30 17:26:51.902 INFO Resolve manifest, link: /ipfs/QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA, subgraph_id: QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA, component: SubgraphRegistrar
graph-node    | Mar 30 17:26:51.907 INFO Resolve schema, link: /ipfs/QmVKMer94nh3p753XmcS5oZEJjVSNTjFLPBKUKR9p2HRwK, subgraph_id: QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA, component: SubgraphRegistrar
graph-node    | Mar 30 17:26:51.909 INFO Resolve data source, source: 6228543, name: Contract, subgraph_id: QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA, component: SubgraphRegistrar
graph-node    | Mar 30 17:26:51.912 INFO Resolve mapping, link: /ipfs/QmRUQ4gb2LErg2RzJzFuAkfg83GULEMEsTid5E8tcsJN75, subgraph_id: QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA, component: SubgraphRegistrar
graph-node    | Mar 30 17:26:51.913 INFO Resolve ABI, link: /ipfs/QmbkUAwCgct32xXhefbrG5oPFN7DQeq3nxJ4ZiRTvzqHGE, name: Contract, subgraph_id: QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA, component: SubgraphRegistrar
graph-node    | Mar 30 17:26:52.174 INFO Set subgraph start block, block_hash: Some(0x950d2051ae85b2859535fbf1530bd407090084839c1a32b1ab1bac6fe91388c1), block_number: Some(6228542), subgraph_id: QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA, component: SubgraphRegistrar
graph-node    | Mar 30 17:26:52.500 INFO Resolve subgraph files using IPFS, subgraph_id: QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA, component: SubgraphAssignmentProvider
graph-node    | Mar 30 17:26:52.504 INFO Resolve manifest, link: /ipfs/QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA, subgraph_id: QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA, component: SubgraphAssignmentProvider
graph-node    | Mar 30 17:26:52.504 INFO Resolve schema, link: /ipfs/QmVKMer94nh3p753XmcS5oZEJjVSNTjFLPBKUKR9p2HRwK, subgraph_id: QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA, component: SubgraphAssignmentProvider
graph-node    | Mar 30 17:26:52.505 INFO Resolve data source, source: 6228543, name: Contract, subgraph_id: QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA, component: SubgraphAssignmentProvider
graph-node    | Mar 30 17:26:52.505 INFO Resolve mapping, link: /ipfs/QmRUQ4gb2LErg2RzJzFuAkfg83GULEMEsTid5E8tcsJN75, subgraph_id: QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA, component: SubgraphAssignmentProvider
graph-node    | Mar 30 17:26:52.505 INFO Resolve ABI, link: /ipfs/QmbkUAwCgct32xXhefbrG5oPFN7DQeq3nxJ4ZiRTvzqHGE, name: Contract, subgraph_id: QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA, component: SubgraphAssignmentProvider
graph-node    | Mar 30 17:26:52.518 INFO Query timing (GraphQL), query_time_ms: 20, variables: {"id":"QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA","skip":0}, query: query deployment($id: ID!, $skip: Int!) { subgraphDeployment(id: $id) { dynamicDataSources(orderBy: id, skip: $skip) { kind network name context source { address abi } mapping { kind apiVersion language file entities abis { name file } blockHandlers { handler filter } callHandlers { function handler } eventHandlers { event handler } } templates { kind network name source { abi } mapping { kind apiVersion language file entities abis { name file } blockHandlers { handler filter } callHandlers { function handler } eventHandlers { event handler } } } } } } , query_id: 53de56a4-7bea-42ef-a11d-8dcae46562fe, subgraph_id: subgraphs, component: GraphQlRunner
graph-node    | Mar 30 17:26:52.522 INFO Successfully resolved subgraph files using IPFS, subgraph_id: QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA, component: SubgraphAssignmentProvider
graph-node    | Mar 30 17:26:52.523 INFO Create attribute indexes for subgraph entities, subgraph_id: QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA, component: SubgraphAssignmentProvider
graph-node    | Mar 30 17:26:52.528 INFO Successfully created attribute indexes for subgraph entities, subgraph_id: QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA, component: SubgraphAssignmentProvider
graph-node    | Mar 30 17:26:52.532 INFO Start subgraph, data_sources: 1, subgraph_id: QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA, component: SubgraphInstanceManager
graph-node    | Mar 30 17:26:52.784 INFO Scanning blocks [6228543, 6228543], range_size: 1, subgraph_id: QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA, component: SubgraphInstanceManager > BlockStream
graph-node    | Mar 30 17:26:54.798 INFO Scanning blocks [6228544, 6228553], range_size: 10, subgraph_id: QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA, component: SubgraphInstanceManager > BlockStream
graph-node    | Mar 30 17:26:55.433 INFO Scanning blocks [6228554, 6228653], range_size: 100, subgraph_id: QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA, component: SubgraphInstanceManager > BlockStream
graph-node    | Mar 30 17:26:55.541 INFO Syncing 1 blocks from Ethereum., code: BlockIngestionStatus, blocks_needed: 1, blocks_behind: 1, latest_block_head: 6229002, current_block_head: 6229001, network_name: rinkeby, component: BlockIngestor
graph-node    | Mar 30 17:26:56.458 INFO 1 trigger found in this block for this subgraph, block_hash: 0xebd31bdf96f92db10edd4af3e3d0e50c47587ccfa9bede147b4bb400e92b6b6c, block_number: 6228577, subgraph_id: QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA, component: SubgraphInstanceManager
graph-node    | Mar 30 17:26:56.463 WARN handleMint: step 1, data_source: Contract, block_hash: 0xebd31bdf96f92db10edd4af3e3d0e50c47587ccfa9bede147b4bb400e92b6b6c, block_number: 6228577, subgraph_id: QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA, component: SubgraphInstanceManager
graph-node    | Mar 30 17:26:56.466 WARN handleMint: step 2, data_source: Contract, block_hash: 0xebd31bdf96f92db10edd4af3e3d0e50c47587ccfa9bede147b4bb400e92b6b6c, block_number: 6228577, subgraph_id: QmYpP1iFhqyuZQf6KMdLvSErY5JWBUQxsTyGdZh1WQ44rA, component: SubgraphInstanceManager
graph-node    | Mar 30 17:27:10.467 INFO Syncing 1 blocks from Ethereum., code: BlockIngestionStatus, blocks_needed: 1, blocks_behind: 1, latest_block_head: 6229003, current_block_head: 6229002, network_name: rinkeby, component: BlockIngestor
graph-node    | Mar 30 17:27:22.993 INFO Syncing 1 blocks from Ethereum., code: BlockIngestionStatus, blocks_needed: 1, blocks_behind: 1, latest_block_head: 6229004, current_block_head: 6229003, network_name: rinkeby, component: BlockIngestor
graph-node    | Mar 30 17:27:38.965 INFO Syncing 1 blocks from Ethereum., code: BlockIngestionStatus, blocks_needed: 1, blocks_behind: 1, latest_block_head: 6229005, current_block_head: 6229004, network_name: rinkeby, component: BlockIngestor
graph-node    | Mar 30 17:27:54.846 INFO Syncing 1 blocks from Ethereum., code: BlockIngestionStatus, blocks_needed: 1, blocks_behind: 1, latest_block_head: 6229006, current_block_head: 6229005, network_name: rinkeby, component: BlockIngestor
graph-node    | Mar 30 17:28:10.262 INFO Syncing 1 blocks from Ethereum., code: BlockIngestionStatus, blocks_needed: 1, blocks_behind: 1, latest_block_head: 6229007, current_block_head: 6229006, network_name: rinkeby, component: BlockIngestor
graph-node
...
```

I have no idea how to debug this since normally Graph prints errors in the log, but here it does nothing. No any following events are processed, no handlers called.

My guess was it's somehow related to infura.io provider. Graph should take blocks on demand from it, and it seems it does. It somehow finds the first trigger block, but then stops. And worse is that the function neither returns, not fails with error. 

Do you have any idea what to do? Maybe I should give it a time to sync from infura.io first? But it founds the trigger, so the block should have enough info to get contract's owner, no?

Help, please.

**Comment by osnwt** [comment]
More info: the same Graph hosted on the `api.thegraph.com` works. So I suspect it's really related to blockchain data provider. Thus, I have other questions:
- Should it work with `infura.io` instead of local blockchain node?
- Should it work with local `geth` with ``--syncmode light``?

**Comment by osnwt** [comment]
Answering myself.

Installed Parity, fully synced it with Rinkeby, connected Graph with INFO log level... Same issue: after finding the 1st trigger there is nothing else happens, no logs, no errors, no data.

Switched to DEBUG log level. In a lot of periodic irrelevant messages I found this:
```
DEBUG web3::transports::http] [739] Sending: {"jsonrpc":"2.0","method":"eth_call","params":[{"data":"0x8da5cb5b","to":"0xa748f237415efb09318c1318d034fb2e87856eb9"},"0x5f7a8e"],"id":739} to http://rinkeby:8545/
```
And this was repeating again and again. Trying to send that from another RPC client, I've got the error **-32000**: `This request is not supported because your node is running with state pruning. Run with --pruning=archive.`
It was obvious that since *'Alchemy uses error -32000 as a timeout'*, Graph treats that error from Parity as timeout as well, and continuously repeats the request without any notice...

Now syncing the blockchain again with this option added...

What's was an unexpected behaviour for me:
- Nowhere is said that, say, Parity node must include such option (it is not the defaults).
- Graph **should not** silently ignore any kind of errors, including -32000. It must, at least, log the failure before repeating the request. Regular timeouts should be fixed with appropriate settings...

Hope this saves someone few hours or days, waisted by me when the same graph worked hosted, but failed locally...

-------

**PS.** infura.io doesn't look like a good option (at least, free account for testing). Graph generates too many requests, so 100K free requests per day are not enough. It could be solved by increasing the poll interval, though.

But for the same `call` it returns me: `"message": "project ID does not have access to archive state"`. So it's unusable for Graph applications where the state history is required.

Running `Geth` in light mode attempts to get the required data but times out with `"no suitable peers available"`, also with error **-32000**...

It seems that either hosted graph solution works, or a local full node with full state history archived is necessary...

**Comment by leoyvens** [comment]
Thanks for the report! I'm surprised by this behaviour, looking at the code we should eventually log this warning when retrying too much `https://github.com/graphprotocol/graph-node/blob/master/chain/ethereum/src/ethereum_adapter.rs#L353`

For subgraphs which use contract calls, we do require Ethereum archive nodes.

**Comment by osnwt** [comment]
I must confess I didn't wait for too long after seen almost instant rescans on other implementations. So perhaps it would log failure finally. Also the requirement for archive nodes is obvious now, however not for new users. I just reported that being unsure what happens, later spending a week syncing the Geth and Parity nodes just to realise that neither light, nor full nodes work without archive option.

So I would only wish you provide more details about requirements (like archive nodes for graphs with calls). But maybe I missed this in documentation being excited with your product! Still, would be good to report at least once a response message from a node on the 1st failure. I I saw that the call is not supported by node, I would find the source of failure. But I just saw that it hangs in the same graph which works well being hosted.

**Comment by That3Percent** [comment]
Closed with #1606. If some part of the system is repeatedly failing (in this case, the `eth_call` method) the log level will be promoted to a warning after 10 successive attempts. This should make it more visible that it may be necessary to intervene to get the system running.

**Comment by cauta** [comment]
@leoyvens sorry because of stupid question, why we need archive node for calling function? while i can read contract data by web3 but the subgraph fail calling it after a week
